### ISSUE TITLE
Error message

### ISSUE THREAD

**Issue by info-anzar** [googlecolab/colabtools#5610]
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LinearRegression
import matplotlib.pyplot as plt
import seaborn as sns

# Load the dataset
# Make sure to upload your CSV file (e.g., Coastal_Tripoli_Metiga_HLLT__50_Numerical-Sheet1-A1:P51-v11.csv)
# to your Google Colab environment before running this code.
df = None # Initialize df to None

try:
    df = pd.read_csv('Coastal_Tripoli_Metiga_HLLT__50_Numerical-Sheet1.csv')
    print("CSV file loaded successfully.")
except FileNotFoundError:
    print("Error: The CSV file 'Coastal_Tripoli_Metiga_HLLT__50_Numerical-Sheet1.csv' was not found.")
    print("Please upload the file to your Colab environment and ensure the name is correct.")
except Exception as e:
    print(f"An unexpected error occurred while loading the CSV file: {e}")

if df is not None:
    # --- Added for debugging: Print column names and data types ---
    print("\nDataFrame Columns and Data Types:")
    print(df.info())
    print("\n")
    # -----------------------------------------------------------

    # Define features (X) and target (y)
    target_column = 'Reserved_channels_in_use'
    flight_level_column = 'Flight_level'

    if target_column not in df.columns:
        print(f"Error: The target column '{target_column}' was not found in the dataset.")
        print("Please check the column name in your CSV file.")
    elif flight_level_column not in df.columns:
        print(f"Error: The column '{flight_level_column}' was not found in the dataset.")
        print("Please check the column name in your CSV file.")
    else:
        # Exclude 'Reserved_channels_in_use' from features
        X = df.drop(target_column, axis=1)
        y = df[target_column]

        # Normalize the dataset
        scaler = StandardScaler()
        X_scaled = scaler.fit_transform(X)
        X_scaled = pd.DataFrame(X_scaled, columns=X.columns)

        # Split the data into training and testing sets
        X_train, X_test, y_train, y_test = train_test_split(X_scaled, y, test_size=0.2, random_state=42)

        # Train a multiple linear regression model
        model = LinearRegression()
        model.fit(X_train, y_train)

        # Make predictions on the test set
        y_pred = model.predict(X_test)

        # Create a DataFrame for plotting predictions
        # Ensure 'Flight_level' is present in X_test before creating predictions_df
        if flight_level_column in X_test.columns:
            predictions_df = pd.DataFrame({
                flight_level_column: X_test[flight_level_column],
                f'Actual_{target_column}': y_test,
                f'Predicted_{target_column}': y_pred
            })

            # Sort by Flight_level for better visualization
            predictions_df = predictions_df.sort_values(by=flight_level_column)

            # Generate a graph showing the prediction of Reserved_channels_in_use (y-axis) against Flight_level (x-axis)
            plt.figure(figsize=(12, 7))
            sns.scatterplot(x=flight_level_column, y=f'Actual_{target_column}', data=predictions_df, label='Actual', alpha=0.7)
            sns.lineplot(x=flight_level_column, y=f'Predicted_{target_column}', data=predictions_df, color='red', label='Predicted')
            plt.title(f'Predicted vs. Actual {target_column.replace("_", " ").title()} by {flight_level_column.replace("_", " ").title()}')
            plt.xlabel(flight_level_column.replace("_", " ").title())
            plt.ylabel(target_column.replace("_", " ").title())
            plt.legend()
            plt.grid(True)
            plt.show()
        else:
            print(f"Warning: '{flight_level_column}' column not found in the test set, skipping prediction plot.")

        print("\nModel Coefficients:")
        print(pd.DataFrame(model.coef_, X.columns, columns=['Coefficient']))
        print(f"\nR-squared on test set: {model.score(X_test, y_test):.2f}")
else:
    print("DataFrame 'df' was not loaded. Cannot proceed with analysis.")

**Comment by info-anzar** [comment]
how to upload a .csv file to my Google Colab environment?

**Comment by cperry-goog** [comment]
<img width="544" height="327" alt="Image" src="https://github.com/user-attachments/assets/3a84263c-60d9-4474-a35b-92caa5464649" />

click on the file icon on the left side and drag the csv in there. open a new bug if you have any issues.
